pkg/metrics/controller.go

```diff
@@ -258,11 +273,66 @@ func (c *Controller) recordGameServerStatusChanges(old, next interface{}) {
 	if newGs.Status.State != oldGs.Status.State {
 		fleetName := newGs.Labels[agonesv1.FleetNameLabel]
 		if fleetName == "" {
-			fleetName = "none"
+			fleetName = defaultFleetTag
 		}
 		recordWithTags(context.Background(), []tag.Mutator{tag.Upsert(keyType, string(newGs.Status.State)),
 			tag.Upsert(keyFleetName, fleetName)}, gameServerTotalStats.M(1))
+
+		// Calculate the duration of the current state
+		duration, err := c.calcDuration(oldGs, newGs)
+		if err != nil {
+			c.logger.Debug(err.Error())
+		} else {
+			recordWithTags(context.Background(), []tag.Mutator{tag.Upsert(keyType, string(oldGs.Status.State)),
+				tag.Upsert(keyFleetName, fleetName)}, gsStateDurationSec.M(duration))
+		}
+	}
+}
+
+// calcDuration - calculate duration between state changes
+// store current time from creationTimestamp for each update received
+// Assumptions: there is a possibility that one of the previous state change timestamps would be evicted,
+// this measure would be skipped
```

Comment:
Should we write something about why we don't care if the item is skipped?

Is it because we're assuming that we'll get enough data across enough GameServers that the occasional miss won't happen?